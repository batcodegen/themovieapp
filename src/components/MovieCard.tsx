import {Image, StyleSheet, Text, View} from 'react-native';
import React, {memo} from 'react';
import {Movie} from '../api/popularmovie';
import HIconTitleView from './HIconTitleView';
import {ENDPOINT} from '../api/endpoints';
import {useTheme} from '@react-navigation/native';

interface MovieCardProp {
  image: string;
  id: string;
  title: string;
  rating: number;
  voteCount: number;
  releaseDate: string;
}

const MovieCard = ({
  title,
  rating,
  voteCount,
  id,
  image,
  releaseDate,
}: MovieCardProp) => {
  const {colors} = useTheme();
  return (
    <View style={styles.cardContainer} key={id}>
      <View style={{flex: 1}}>
        <Image
          source={{uri: `${ENDPOINT.IMAGE_BASE_URL}/w500${image}`}}
          style={styles.cardImage}
          resizeMode="contain"
        />
      </View>
      <View style={{flex: 1, justifyContent: 'flex-start'}}>
        <Text
          style={[styles.cardTitle, {color: colors.text}]}
          numberOfLines={2}
          ellipsizeMode="tail">
          {title}
        </Text>
      </View>
      <View
        style={{
          flex: 1,
          justifyContent: 'flex-end',
        }}>
        <HIconTitleView
          title={`${rating.toFixed(1)} (${voteCount})`}
          iconColor="#fdc434"
          iconName={'star'}
        />
        <HIconTitleView title={releaseDate} iconName="calendar" />
      </View>
    </View>
  );
};

export default memo(MovieCard);

const styles = StyleSheet.create({
  cardContainer: {flex: 1, margin: 10},
  cardImage: {width: '100%', borderRadius: 10, aspectRatio: 2 / 3},
  cardTitle: {
    fontWeight: 'bold',
    fontSize: 16,
    marginTop: 8,
    marginBottom: 5,
    textAlign: 'left',
  },
  cardStars: {fontSize: 12, marginTop: 8, marginBottom: 1},
  cardDate: {fontSize: 12, marginTop: 1, marginBottom: 1},
});
