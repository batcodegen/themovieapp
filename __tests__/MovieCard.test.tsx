import React from 'react';
import {render, screen} from '@testing-library/react-native';
import MovieCard from '../src/components/MovieCard';

const props = {
  id: '1',
  image: 'https://example.com',
  rating: 4.5,
  releaseDate: '2024-01-20',
  title: 'Minions',
  voteCount: 123,
};

describe('Header Component', () => {
  it('Renders correctly', () => {
    render(<MovieCard {...props} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
