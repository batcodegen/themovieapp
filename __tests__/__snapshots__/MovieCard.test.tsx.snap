// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "uri": "https://image.tmdb.org/t/p/w500https://example.com",
        }
      }
      style={
        {
          "aspectRatio": 0.6666666666666666,
          "borderRadius": 10,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        [
          {
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 5,
            "marginTop": 8,
            "textAlign": "left",
          },
          {
            "color": undefined,
          },
        ]
      }
    >
      Minions
    </Text>
  </View>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "marginTop": 2,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#fdc434",
              "fontSize": 16,
            },
            undefined,
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          [
            {
              "marginStart": 5,
              "textAlign": "left",
            },
            {
              "color": undefined,
            },
          ]
        }
      >
        4.5 (123)
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginTop": 2,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 16,
            },
            undefined,
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          [
            {
              "marginStart": 5,
              "textAlign": "left",
            },
            {
              "color": undefined,
            },
          ]
        }
      >
        2024-01-20
      </Text>
    </View>
  </View>
</View>
`;
